core/primitives/src/rpc.rs

```diff
@@ -12,6 +12,38 @@ use crate::views::{
     ExecutionOutcomeWithIdView, LightClientBlockLiteView, QueryRequest, StateChangeWithCauseView,
     StateChangesKindsView, StateChangesRequestView,
 };
+use serde::de::DeserializeOwned;
+use serde_json::Value;
+
+#[derive(Serialize)]
+pub struct RpcParseError(pub String);
```

Comment:
Let's extract this file from core primitives and merge it into `near-jsonrpc-primitives`